portal.ejb/src/main/java/org/vamdc/portal/session/queryLog/QueryStore.java

```diff
@@ -0,0 +1,223 @@
+		this.processing = processing;
+	}
+
+	/**
+	 * return result of querystore for a request
+	 * 
+	 * @return
+	 **/	
+	public QueryStoreResponse getNodeUuid(String node) {
+		// not requested yet
+		if (this.uuids.containsKey(node) == false)
+			return new QueryStoreResponse(QueryStoreResponse.STATUS_UNKNOWN,
+					"", "");
+
+		return this.uuids.get(node);
+	}		
+
+	/**
+	 * Ask a permanent identifier for the request identified by the token string
+	 * 
+	 * @param token  token of a head request
+	 */
+	//@Asynchronous
+	public void associateRequest(String token, String node) {
+		this.processing = "Starting process";
+		QueryStoreResponse result = null;
+		Integer count = 0;
+		try {
+			String request = this.getRequest(token, this.getUserEmail(),
+					this.getIpAdress());
+						// not tried yet or previous try failed
+			if (!uuids.containsKey(node)
+					|| !uuids.get(node).getStatus()
+							.equals(QueryStoreResponse.STATUS_SUCCESS)) {	
+				while ((result == null || QueryStoreResponse.STATUS_EMPTY.equals(result.getStatus())) && count < this.retry) {
+					result = this.doRequest(request);
+					Thread.sleep(this.retryInterval);
+					count++;
+				}
+				this.uuids.put(node, result);
+			}
+		} catch (Exception e) {
+			log.debug(e);
+		}
+		this.processing = "INFO : A  request was sent to the query store to get a citation link. Request status :  " + uuids.get(node).getStatus();
+	}
+
+
+	/**
+	 * Send an HTTP request to the query store to get a UUID corresponding to a
+	 * head token
+	 * 
+	 * @param token
+	 *            token of a head request
+	 * @param email
+	 *            email of registered user
+	 * @param userIp
+	 *            ip of user asking
+	 * @return
+	 * @throws ClientProtocolException
+	 * @throws IOException
+	 * @throws HttpException
+	 * @throws KeyStoreException 
+	 * @throws NoSuchAlgorithmException 
+	 * @throws KeyManagementException 
+	 */
+	private QueryStoreResponse doRequest(String requestString)
+			throws IOException, HttpException, NoSuchAlgorithmException, KeyStoreException, KeyManagementException {
+		
+		RequestConfig requestConfig = RequestConfig.custom()
+				.setConnectTimeout(3 * 1000).build();		
+		
+		HttpClient httpClient = HttpClientBuilder.create()
+				.setDefaultRequestConfig(requestConfig).build();
+				
+		HttpGet request = new HttpGet(requestString);	
+		request.addHeader("User-Agent", Settings.PORTAL_USER_AGENT.get());
+		StringBuilder result = new StringBuilder();			
+		
+		//SSLHandShakeException (IOException) if missing certificate
+		HttpResponse response = httpClient.execute(request);	
+		
+		if (response.getStatusLine().getStatusCode() == 200) {
+			BufferedReader rd = new BufferedReader(new InputStreamReader(
+					response.getEntity().getContent()));
+
+			String uuid;
+			while ((uuid = rd.readLine()) != null) {
+				result.append(uuid);
+			}
+			rd.close();
+			
+		} else {
+			// empty response
+			if (response.getStatusLine().getStatusCode() == 204) {
```

Comment:
wondering if `response.getStatusLine().getStatusCode()` should be extracted into a local variable